tools/jdbc/src/main/java/org/duckdb/DuckDBDatabaseMetaData.java

```diff
@@ -83,13 +83,12 @@ public String getDatabaseProductName() throws SQLException {
 
 	@Override
 	public String getDatabaseProductVersion() throws SQLException {
-		Statement s = conn.createStatement();
-		ResultSet rs = s.executeQuery("PRAGMA version");
-		rs.next();
-		String result = rs.getString(1);
-		rs.close();
-		s.close();
-		return result;
+		try (Statement s = conn.createStatement();
+			ResultSet rs = s.executeQuery("PRAGMA version")) {
+			rs.next();
+			String result = rs.getString(1);
```

Comment:
Should the resultSet not be closed as well here?

Reply:
Indeed it should be, and it is. It's included in the _resource list_ passed to the `try`, as the `Statement` line separated from the `ResultSet` by `;`. This passes 2 resources to the `try`, and they'll be be closed in reverse-initialization order.

https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html

Reply:
Thanks - I had missed the semicolon there. Not too experienced with Java :)

Reply:
👍 thanks for reviewing!